code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -94,6 +66,23 @@
 		return 0
 	return ..()
 
+/obj/item/weapon/gun/energy/laser/proc/downgradelaser(var/mob/living/carbon/human/M)
+	switch(projectile_type)
+		if("/obj/item/projectile/beam/heavylaser")
+			projectile_type = /obj/item/projectile/beam
+			fire_sound = 'sound/weapons/Laser.ogg'
+		if("/obj/item/projectile/beam/captain", /obj/item/projectile/beam, "/obj/item/projectile/beam", /obj/item/projectile/beam/retro)
+			projectile_type = /obj/item/projectile/beam/lightlaser
+		if(/obj/item/projectile/beam/lightlaser)
+			projectile_type = /obj/item/projectile/beam/weaklaser
+		if(/obj/item/projectile/beam/weaklaser)
+			projectile_type = /obj/item/projectile/beam/veryweaklaser
+	in_chamber = null
+	in_chamber = new projectile_type(src)
+	fire_delay +=3
+	to_chat(M, "<span class='warning'>Something inside \the [src] pops.</span>")
+	return
```

Comment:
A trailing return is a null return at the very end of a proc.

They're bad because they're completely unnecessary. By default, procs return null when they reach their end, so putting a `return` at the end of the proc is redundant.